ncpi_fhir_utility/update_versions.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+"""
+Updates versions and dates in all of our FHIR resources.
+
+Example:
+    $ scripts/update_versions.py 1.1.1
+"""
+import json
+import sys
+from datetime import datetime, timezone
+from pathlib import Path
+
+import click
+
+
+def bump_ig_version(filePath, new_version):
+    with filePath.open(mode="r") as f:
+        body = [
+            line
+            if not line.startswith("version ")
+            else f"version = {new_version}\n"
+            for line in f
+        ]
+    with filePath.open(mode="w") as f:
+        f.writelines(body)
+
+
+def bump_resource_version(filePath, new_version, our_urls):
+    try:
+        with filePath.open(mode="r") as f:
+            body = json.load(f)
+    except json.decoder.JSONDecodeError:
+        return
+    if body.get("url", "").startswith(our_urls):
+        body["version"] = new_version
+        if "date" in body:
+            body["date"] = datetime.now().astimezone(timezone.utc).isoformat()
+        with filePath.open(mode="w") as f:
+            json.dump(body, f, indent=2)
+
+
+def run(site_root, new_version, our_urls):
+    our_urls = tuple(s.strip() for s in our_urls.split(","))
+    site_root = Path(site_root)
+    if not site_root.is_dir():
+        print(
+            f"Error: <site_root> dir '{site_root}' must exist.", file=sys.stderr
+        )
+    ini_file = Path(site_root, "ig.ini")
+    if not ini_file.is_file():
+        print(f"Error: <ini_file> '{ini_file}' must exist.", file=sys.stderr)
```

Comment:
I think this should be able to use the logger?

Reply:
Yes but why?

Reply:
√